lib/api/BackbeatAPI.js

```diff
@@ -0,0 +1,205 @@
+'use strict'; // eslint-disable-line strict
+
+const async = require('async');
+const zookeeper = require('node-zookeeper-client');
+
+const { errors } = require('arsenal');
+
+const BackbeatProducer = require('../BackbeatProducer');
+const routes = require('./routes');
+
+// In-Sync Replicas
+const ISRS = 3;
+
+/**
+ * Class representing Backbeat API endpoints and internals
+ *
+ * @class
+ */
+class BackbeatAPI {
+    /**
+     * @constructor
+     * @param {object} config - configurations for setup
+     * @param {werelogs.Logger} logger - Logger object
+     */
+    constructor(config, logger) {
+        this._zkConfig = config.zookeeper;
+        this._repConfig = config.extensions.replication;
+        this._crrTopic = this._repConfig.topic;
+        this._crrStatusTopic = this._repConfig.replicationStatusTopic;
+        this._metricsTopic = config.metrics.topic;
+        this._queuePopulator = config.queuePopulator;
+        this._kafkaHost = config.kafka.hosts;
+        this._logger = logger;
+
+        this._crrProducer = null;
+        this._crrStatusProducer = null;
+        this._metricProducer = null;
+        this._zkClient = null;
+    }
+
+    /**
+     * Check if incoming request is valid
+     * @param {string} route - request route
+     * @return {boolean} true/false
+     */
+    isValidRoute(route) {
+        return routes.map(route => route.path).includes(route);
```

Comment:
Can use `some()` method here instead of mapping and then checking the resulting mapped array.